miner/bid_simulator.go

```diff
@@ -442,10 +440,14 @@ func (b *bidSimulator) clearLoop() {
 func (b *bidSimulator) sendBid(_ context.Context, bid *types.Bid) error {
 	timer := time.NewTimer(1 * time.Second)
 	defer timer.Stop()
+
+	replyCh := make(chan error)
+	b.newBidCh <- newBidPackage{bid: bid, feedback: replyCh}
+	b.AddPending(bid.BlockNumber, bid.Builder, bid.Hash())
+
 	select {
-	case b.newBidCh <- bid:
-		b.AddPending(bid.BlockNumber, bid.Builder, bid.Hash())
-		return nil
+	case reply := <-replyCh:
+		return reply
 	case <-timer.C:
 		return types.ErrMevBusy
```

Comment:
Line 445 may be blocked when newBidCh is busy, how about using two select to avoid it
```
replyCh := make(chan error)

select {
case b.newBidCh <- newBidPackage{bid: bid, feedback: replyCh}:
b.AddPending(bid.BlockNumber, bid.Builder, bid.Hash())
case <-timer.C:
return types.ErrMevBusy
}

select {
case reply := <-replyCh:
return reply
case <-timer.C:
return types.ErrMevBusy
}
```